Sp 10 registration page

nenovKrasimir opened the pull request.

nenovKrasimir committed:
Edited users table - added unique username

nenovKrasimir committed:
Update README.md

nenovKrasimir committed:
Update README.md

added instructions for MailTraip (Email Sending)

nenovKrasimir committed:
created verify email template

nenovKrasimir committed:
created AuthController for registering users and sending verification emails + handling logic for verifying user

nenovKrasimir committed:
added api rotues for creating users & verifying email

nenovKrasimir committed:
added complex password validator

nenovKrasimir committed:
created register mail

nenovKrasimir committed:
making available routes for react

nenovKrasimir committed:
edited template for email verifying

nenovKrasimir committed:
created basic components for handling verification and register logic

nenovKrasimir committed:
added css files for components

nenovKrasimir committed:
created register mail and edited welcome blade

nenovKrasimir committed:
Update README.md

added information for mailtrap

nenovKrasimir committed:
Update README.md

realestateSoftuni commented on app/Http/Controllers/AuthController.php:
```diff
@@ -0,0 +1,54 @@
+    public function create_user(Request $request)
+    {
+        request()->validate([
+            'username'=>'required|unique:users',
```
I think is good idea to add |max:255|string|

nenovKrasimir replied:
Done

realestateSoftuni commented on app/Http/Controllers/AuthController.php:
```diff
@@ -0,0 +1,54 @@
+        if(!empty($user))
+        {
+            $user->email_verified_at = date('Y-m-d H:i:s');
+            $user->remember_token = Str::random(40);
```
maybe 20 characters is enough

nenovKrasimir replied:
Done

realestateSoftuni commented on app/Http/Controllers/AuthController.php:
```diff
@@ -0,0 +1,54 @@
+        }
+        else
+        {
+            abort(404);
```
We can log some message here

nenovKrasimir replied:
Done

realestateSoftuni commented on resources/js/components/EmailVerify.jsx:
```diff
@@ -0,0 +1,42 @@
+    const { token } = useParams();
+    const [loading, setLoading] = useState(true);
+    const [message, setMessage] = useState('');
+
```
thanks for the comments is good practice

nenovKrasimir replied:
Done -> Added more comments.

realestateSoftuni commented on resources/js/components/NotFound.jsx:
```diff
@@ -0,0 +1,13 @@
+// NotFound.jsx
+import React from 'react';
+
```
We can use some cool template for this page , https://colorlib.com/wp/free-404-error-page-templates/   this site is only for example fell free to use other

realestateSoftuni commented on resources/js/components/UserRegister.jsx:
```diff
@@ -0,0 +1,123 @@
+        setIsLoading(true);
+
+        try {
+            const response = await axios.post('http://localhost:8000/api/create_user', formData);
```
localhost? I think we need to set variable and change it for the different environments

nenovKrasimir replied:
Done

nenovKrasimir committed:
Update EmailVerify.jsx

added more comments

nenovKrasimir committed:
Update AuthController.php

added max string 255 + abort message

nenovKrasimir committed:
Update UserRegister.jsx

edited api address (removed localhost and loading the address dynamically thru script from welcome.blade.php

nenovKrasimir committed:
Update welcome.blade.php

added script for loading app url from the laravel env file

realestateSoftuni commented:
Now evriting look good thanks i will merge it after home page is ready
